modules/squarepine_audio/effects/NotifiableAudioParameterFloat.h

```diff
@@ -0,0 +1,46 @@
+class NotifiableAudioParameterFloat : public AudioParameterFloat
```

Comment:
It's worth mentioning that this class is quite redundant when you could be using a `juce::AudioProcessorValueTreeState` in place which inherently controls a `juce::ValueTree`, populated with parameters for the specific processor.

A fair warning that the levels of indirection needed to listen for parameters in this new structure will just bloat your code into something tedious to extend and manage.

If the argument is that `InternalProcessor` needs to be provide some higher level abstraction surrounding processors so as to create multiple processors from a single APVTS, then I understand that and could give it some thought. And really that would be a valid feature request to JUCE.

If you look at [InternalProcessor::getAPVTS()](https://github.com/SquarePine/squarepine_core/blob/28d2c1609d7ea807ccbefef976af19a3e7169f23/modules/squarepine_audio/core/InternalProcessor.h#L57), you'll see that each processor currently supports storing and controlling its own APVTS. Each of which can be configured for your processors